web/tests/test_tournament.py

```diff
@@ -4,52 +4,296 @@
+        url = self.tournament_endpoint + \
+            str(self.tournament_1.id) + \
+            "/player/" + str(self.tournament_1_player.id) + \
+            "/delete/"
+        before = TournamentPlayer.query.count()
+        self.assertEqual(1, before)
+        response = self.client.post(url)
+        after = TournamentPlayer.query.count()
+        self.assertEqual(0, after)
+
+    # new player - post
+    def test_can_create_new_tournament_player(self):
+        url = self.tournament_endpoint + \
+            str(self.tournament_1.id) + \
+            "/player/new/"
+        before = TournamentPlayer.query.count()
+        self.assertEqual(1, before)
+        response = self.client.post(
+            url,
+            data={ "name": 'player 2',
+                   "aga_num": '23456',
+                   "rating": '11',
+                   "affiliation": 'The Go Getters',
+                   "state": 'WY',
+                   "address": '101 Broadway',
+                   "email": 'player2@example.com',
+                   "phone": '234-567-8901',
+                   "citizenship": 'usa',
+                   "dob": '11/11/11',
+                  "tournament_id": self.tournament_1.id })
+        self.assertEqual(302, response.status_code)
+        after = TournamentPlayer.query.count()
+        self.assertEqual(2, after)
+        new_player = TournamentPlayer.query.all()[-1]
+        self.assertEqual(new_player.name, 'player 2')
+
+    #  edit player - post
+    def test_it_can_edit_an_existing_tournament_player(self):
+        url = self.tournament_endpoint + \
+            str(self.tournament_1.id) + \
+            "/player/" + str(self.tournament_1_player.id) + \
+            "/edit/"
+        player = TournamentPlayer.query.get(self.tournament_1_player.id)
+        player.name = "Test Edit"
+        response = self.client.post(
+            url,
+            data={'name': player.name,
+                  "aga_num": player.aga_num,
+                  "rating": player.rating,
+                    "affiliation": player.affiliation,
+                    "state": player.state,
+                    "address": player.address,
+                    "email": player.email,
+                    "phone": player.phone,
+                    "citizenship": player.citizenship,
+                    "dob": player.dob })
+        player = TournamentPlayer.query.get(self.tournament_1_player.id)
+        self.assertEqual("Test Edit", player.name)
+
+    # sad path - cannot create, edit, or delete player if tournament submitted
+    def test_it_cannot_add_new_player_if_tournament_marked_submitted(self):
+        tournament_edit_endpoint = self.tournament_endpoint + \
+            str(self.tournament_1.id) + '/edit'
+        t = Tournament.query.first()
+        #  mark tournament as submitted
+        response = self.client.post(
+            '/tournament/'+ str(t.id)+'/edit/', #  url_for()
+            data={"event_name": t.event_name,
```

Comment:
These tests are a little wordy. since you're constructing a big tournament object each time. How about a helper func